### ISSUE TITLE
What should Clear do? (and deleting uploaded files)

### ISSUE THREAD

**Issue by joyously** [mavoweb/mavo#275]
While trying to see if a storage backend needs to handle deleting uploaded files, I read this code in the `clear` function.
https://github.com/mavoweb/mavo/blob/1281b72b2154d9171676c7c5d87cf12a00b2887b/src/mavo.js#L322

At face value, it looks fine  _(actually the `store` function does not take a parameter)_, but should `store` be called here? Shouldn't the data only be saved when the user chooses `Save`?

It did not help me figure out if I need to delete an uploaded file. How is that handled?

**Comment by LeaVerou** [comment]
It's up to the implementation. Either storing an empty file or deleting the file would both be acceptable behaviors. Currently, AFAIK none of the existing file-based storage backends delete the file. There is no way to do so via Mavo.

**Comment by joyously** [comment]
Your answer sounds like it is referring to the saved data that some backends store as files. 
My question was about uploaded files, which is part of the data that is being stored.
Additionally, it is about whether storing data before clearing it is the expected sequence. I know if I edit something and then click Clear, I do not want my edits saved first, or I would have clicked Save.

**Comment by LeaVerou** [comment]
You're right, Clear shouldn't store data before clearing, that's against expectation, and could be quite catastrophic. Btw Clear was moved to a plugin yesterday: https://plugins.mavo.io/plugin/clear
I should still fix this there.
